src/Microsoft.ML.Data/Scorers/RowToRowScorerBase.cs

```diff
@@ -143,7 +143,7 @@ public override IRowCursor[] GetRowCursorSet(out IRowCursorConsolidator consolid
             var inputs = Source.GetRowCursorSet(out consolidator, predicateInput, n, rand);
             Contracts.AssertNonEmpty(inputs);
 
-            if (inputs.Length == 1 && n > 1 && WantParallelCursors(predicate))
+            if (inputs.Length == 1 && n > 1 && WantParallelCursors(predicate) && Source.GetRowCount() != 1)
```

Comment:
>GetRowCount [](start = 88, length = 11)

However, the issue with this solution is that if user sets up a true enumerable, we will do a parallel clustering setup.  At least emprirically it appears that setting up parallel cursors for scoring is 10k x slower vs not. Why we even doing this at all?  The only time it will be worth doing if we are having on the order of 10k items to score.

Reply:
We don't know size of Source in general, so nothing we can do. At least with current interfaces. I guess I can propagate parameter which would forbid usage of parallel cursor, but's it's a lot of changes.
Plus in current situation, we are forcing data to be array [https://github.com/dotnet/machinelearning/blob/3012428bdf2629c991f1dd90eac7d424b20a2fcf/src/Microsoft.ML/PredictionModel.cs#L161](https://github.com/dotnet/machinelearning/blob/3012428bdf2629c991f1dd90eac7d424b20a2fcf/src/Microsoft.ML/PredictionModel.cs#L161)

---
In reply to: [189992635](https://github.com/dotnet/machinelearning/pull/197#discussion_r189992635) [](ancestors = 189992635)

Reply:
Specifically that array can possibly be recycled.
Longer term we should use the row to row mappers that Yael has developed.

---
In reply to: [190086316](https://github.com/dotnet/machinelearning/pull/197#discussion_r190086316) [](ancestors = 190086316,189992635)